### PR TITLE
[MOB-1006] Add support for 'read' flag from getMessages API

### PR DESCRIPTION
**Pull request by Ayyanchira**
Added inbox parameter to constructor. Default read status will be false unless specified in json obtained from server.
One message in json file used for testing updated to have read status as true.
One Test method updated and one test method added to verify the functionality.

### CODE REVIEW

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java [hunk 1]
```diff
@@ -289,18 +290,18 @@ static IterableInAppMessage fromJSONObject(JSONObject messageJson, IterableInApp
         }
 
         Boolean saveToInbox = messageJson.has(IterableConstants.ITERABLE_IN_APP_SAVE_TO_INBOX) ? messageJson.optBoolean(IterableConstants.ITERABLE_IN_APP_SAVE_TO_INBOX) : null;
+        Boolean read = messageJson.has(IterableConstants.ITERABLE_IN_APP_READ) ? messageJson.optBoolean(IterableConstants.ITERABLE_IN_APP_READ) : false;
```
Comment (vbabenkoru) [inline iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: The code in this file is identical to the original in behavior. If I revert the changes in this file, the test still passes. Looks like it was supporting reading `read` from the server already.

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableInboxTest.java [hunk 2]
```diff
@@ -92,17 +92,27 @@ public void testSetRead() throws Exception {
...
+        assertTrue(inboxMessages.get(1).isRead());
+        inAppManager.setRead(inboxMessages.get(0), true);
+        assertEquals(0, inAppManager.getUnreadInboxMessagesCount());
+        assertEquals(2, inAppManager.getInboxMessages().size());
+        assertTrue(inboxMessages.get(0).isRead());
+        assertTrue(inboxMessages.get(1).isRead());
+    }
 
+    @Test
+    public void testMessageReadStatus() throws Exception {
```
Comment (vbabenkoru) [inline iterableapi/src/test/java/com/iterable/iterableapi/IterableInboxTest.java]: I think this should indicate that it's testing read status from the server, right now it's not clear what the difference between the two test methods is.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java (hunk 1 above)
Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: Yep, we had all the provision already. I realized when I went to create constants! 😅
Thought of having the value assigned in constructor though. I didn't realize however that the test would pass without this change too!

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableInboxTest.java (hunk 2 above)
Comment (Ayyanchira) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInboxTest.java]: I felt this repeated too. But because the other method was more focused on `setRead` method, I thought of having it seperate. We can remove it altogether and just verify read status in already existing test method.

Comment (vbabenkoru) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableInboxTest.java]: It's a good idea to have 2 separate tests for this, they just need to have names to tell what the difference between them is.

File: iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java (hunk 1 above)
Comment (vbabenkoru) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: This change basically moved `read` from assignment to a constructor parameter, but the end result is the same.

Comment (Ayyanchira) [reply iterableapi/src/main/java/com/iterable/iterableapi/IterableInAppMessage.java]: Ok. I will revert this one.
